src/jit/codegenxarch.cpp

```diff
@@ -8134,10 +8109,11 @@ void CodeGen::genStoreRegToStackArg(var_types type, regNumber srcReg, int offset
 //    For non tail calls this is the outgoingArgSpace.
 void CodeGen::genPutStructArgStk(GenTreePutArgStk* putArgStk)
 {
-    var_types targetType = putArgStk->TypeGet();
+    GenTree*  source     = putArgStk->gtGetOp1();
+    var_types targetType = source->TypeGet();
 
 #if defined(_TARGET_X86_) && defined(FEATURE_SIMD)
-    if (targetType == TYP_SIMD12)
+    if (varTypeIsSIMD(targetType) && (putArgStk->gtNumSlots == 3))
```

Comment:
You may want to add a function that handles this re-occurring predicate:

IsSIMD12(GenTreePutArgStk* source)